src/main/java/com/github/legoatoom/connectiblechains/chain/ChainTypesRegistry.java

```diff
@@ -0,0 +1,94 @@
+package com.github.legoatoom.connectiblechains.chain;
+
+import com.github.legoatoom.connectiblechains.ConnectibleChains;
+import com.github.legoatoom.connectiblechains.util.Helper;
+import com.mojang.serialization.Lifecycle;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import net.fabricmc.fabric.api.event.registry.FabricRegistryBuilder;
+import net.fabricmc.fabric.api.event.registry.RegistryAttribute;
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.DefaultedRegistry;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public class ChainTypesRegistry {
+    public static final Identifier DEFAULT_CHAIN_TYPE_ID = Helper.identifier("iron_chain");
+    public static final RegistryKey<? extends Registry<ChainType>> CHAIN_TYPE_KEY =
+            RegistryKey.ofRegistry(Helper.identifier("chain_types"));
+    public static final DefaultedRegistry<ChainType> REGISTRY = FabricRegistryBuilder.from(
+                    new DefaultedRegistry<>(DEFAULT_CHAIN_TYPE_ID.toString(), CHAIN_TYPE_KEY, Lifecycle.stable()))
+            .attribute(RegistryAttribute.SYNCED).buildAndRegister();
+
+    public static final Map<Item, ChainType> ITEM_CHAIN_TYPES = new Object2ObjectOpenHashMap<>(64);
+
+    public static final ChainType DEFAULT_CHAIN_TYPE = register(DEFAULT_CHAIN_TYPE_ID, Items.CHAIN);
+    @SuppressWarnings("unused")
+    public static final ChainType IRON_CHAIN = DEFAULT_CHAIN_TYPE;
+    private static boolean locked = false;
+
+    /**
+     * Prevents registration with {@link #registerDynamic(Item)}
+     */
+    public static void lock() {
+        locked = true;
+    }
+
+    /**
+     * Used to register chain types on initialization. Cannot register a type twice.
+     * @param item the chain type's item
+     * @return the new {@link ChainType}
+     */
+    @SuppressWarnings("UnusedReturnValue")
+    public static ChainType register(Item item) {
+        Identifier id = Registry.ITEM.getId(item);
+        if (id == Registry.ITEM.getDefaultId()) {
+            throw new AssertionError("Cannot create chain type with unregistered item");
+        }
+        if (REGISTRY.containsId(id)) return REGISTRY.get(id);
```

Comment:
Again, throwing errors is not that good, use logging of errors and then skip.

Reply:
Now prints an error and returns the default type